dvb/datascience/pipeline.py

```diff
@@ -21,6 +24,63 @@ class Status(Enum):
     FINISHED = 3
 
 
+class TransformOutputsBase(metaclass=abc.ABCMeta):
+
+    @abc.abstractmethod
+    def save_pipe_output(self, pipe_name: str, transform_nr: int, output: Dict[str, Any]) ->None:
+        """save the pipe output"""
+
+    @abc.abstractmethod
+    def load_pipe_output(self, pipe_name: str, transform_nr: int) ->  Dict[str, Any]:
+        """load the pipe outputs, returns None when the output does not exists"""
+
+    @abc.abstractmethod
+    def contains_pipe(self, pipe_name: str, transform_nr: int) -> bool:
+        """returns True when the pipe with pipe_name has stored his output.
+        For some implementations this is more efficient as checking whether load_pipe_output is not None
+        """
+
+class TransformOutputsMemory(TransformOutputsBase):
+
+    # outputs = None
+
+    def __init__(self):
+        self.outputs: Dict[Tuple[str, int], Dict[str, Any]] = {}
+
+    def save_pipe_output(self, pipe_name: str, transform_nr: int, output: Dict[str, Any]) ->None:
```

Comment:
```suggestion
    def save_pipe_output(self, pipe_name: str, transform_nr: int, output: Dict[str, Any]) -> None:
```
(space after the arrow)